### DISCUSSION TITLE
Collection objects not sorted / indexed the way they are ordered in the outliner

### DISCUSSION THREAD

**Discussion by pr0jectile** [JacquesLucke/animation_nodes#1819] [Q&A]
What am I doing here? The viewer is showing the objects in a different order (i.e. not going from 21 , 22 ,23 etc).

I've followed the guide on the MIDI documentation page but it doesn't seem to work properly.

<img width="1570" alt="Screenshot 2021-12-18 at 4 34 55 pm" src="https://user-images.githubusercontent.com/12081053/146648682-415c690e-3da7-4ef8-b235-b85d0d8aeb67.png">

**Comment by OmarEmaraDev** [comment]
You need to record the needed order as described here:
https://docs.animation-nodes.com/release_notes/v2_2/general/#id-keys
